source/frontend/src/features/mapSideBar/lease/tabs/consultations/edit/models.ts

```diff
@@ -0,0 +1,93 @@
+import { fromApiOrganization, fromApiPerson, IContactSearchResult } from '@/interfaces';
+import { ApiGen_Concepts_ConsultationLease } from '@/models/api/generated/ApiGen_Concepts_ConsultationLease';
+import { ApiGen_Concepts_Organization } from '@/models/api/generated/ApiGen_Concepts_Organization';
+import { ApiGen_Concepts_Person } from '@/models/api/generated/ApiGen_Concepts_Person';
+import { getEmptyBaseAudit } from '@/models/defaultInitializers';
+import { exists } from '@/utils';
+import { emptyStringtoNullable, toNullableId, toTypeCodeNullable } from '@/utils/formUtils';
+
+export class ConsultationFormModel {
+  public id: number;
+  public leaseId: number;
+  public contact: IContactSearchResult;
+  public primaryContactId: number;
+  public consultationTypeCode: string;
+  public consultationTypeDescription: string;
+  public consultationStatusTypeDescription: string;
+  public otherDescription: string;
+  public requestedOn: string;
+  public isResponseReceived: boolean | null;
+  public responseReceivedDate: string;
+  public comment: string;
+  public rowVersion: number;
+
+  constructor(leaseId: number) {
+    this.leaseId = leaseId;
+    this.id = 0;
+    this.consultationTypeCode = '';
+    this.consultationTypeDescription = '';
+    this.consultationStatusTypeDescription = '';
+    this.otherDescription = '';
+    this.requestedOn = '';
+    this.isResponseReceived = null;
+    this.responseReceivedDate = '';
+    this.comment = '';
+    this.rowVersion = 0;
+  }
+
+  static fromApi(
+    apiModel: ApiGen_Concepts_ConsultationLease,
+    person: ApiGen_Concepts_Person | null,
+    organization: ApiGen_Concepts_Organization | null,
+  ): ConsultationFormModel {
+    const consultation = new ConsultationFormModel(apiModel.leaseId);
+    consultation.id = apiModel.id;
+    consultation.consultationTypeCode = apiModel.consultationTypeCode?.id ?? '';
+    consultation.consultationTypeDescription = apiModel.consultationTypeCode?.description ?? '';
+    consultation.consultationStatusTypeDescription =
+      apiModel.consultationStatusTypeCode?.description ?? '';
+    consultation.otherDescription = apiModel.otherDescription ?? '';
+    consultation.requestedOn = apiModel.requestedOn ?? '';
+    consultation.isResponseReceived = apiModel.isResponseReceived ?? null;
+    consultation.responseReceivedDate = apiModel.responseReceivedDate ?? '';
+    consultation.comment = apiModel.comment ?? '';
+    consultation.rowVersion = apiModel.rowVersion ?? 0;
+
+    const contact: IContactSearchResult | undefined = exists(person)
+      ? fromApiPerson(person)
+      : exists(organization)
+      ? fromApiOrganization(organization)
+      : undefined;
+
+    consultation.primaryContactId = apiModel.primaryContactId;
```

Comment:
hmm, I think this is ok - just want to make sure we shouldn't be conditionally setting this based on whether the id is an org?

Reply:
AFAIK only persons can be primary contacts